### PR TITLE
Re-enable XCUITest

### PR DESCRIPTION
**Pull request by xianwen**
**Summary**
Both `tools/xcuitest/ExampleApp.xctestrun` and `tools/xcuitest/XCUITests-Runner.app.zip` are reusable, and their sizes are small.
Both were generated by me locally via Xcode after a manual XCUITest run.

1. I heavily modified the `ExampleApp.xctestrun` file, so that it can support simply putting everything under the `$(UI_TESTS_TMP)` dir. Otherwise, it would expect many more artifacts to be presented and expects a very different directory layout.
2. For the `XCUITests-Runner.app.zip` I removed the `XCUITest.xctest` file bundled in it since it's generated during every test run, and the `xctestrun` file have been updated to reflect the new location of the `XCUITest.xctest` file

**Verified on CircleCI:**
https://app.circleci.com/pipelines/github/airbnb/BuckSample/97/workflows/a6fe4beb-5bbb-45ed-a0e2-2ef89fb6eff0/jobs/96/parallel-runs/0/steps/0-103

**Test Log:**
```
Testing started on 'iPhone 11 Pro'
2020-08-31 23:19:44.668 xcodebuild[2145:42679]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
/Users/distiller/Library/Developer/Xcode/DerivedData/temporary-hgngnfbtbdnqacfiifcsibormbbe/Logs/Test/Test-Transient Testing-2020.08.31_23-19-42-+0000.xcresult/Staging/1_Test/Diagnostics/XCUITests-2511D29C-2BCA-4792-B0AF-7EDE50A5DEFA/XCUITests-DC737008-8C95-48B3-BFFA-2993922354D6/Session-XCUITests-2020-08-31_231944-0wQ37D.log
2020-08-31 23:19:44.670 xcodebuild[2145:36022] [MT] IDETestOperationsObserverDebug: (A8CF1B79-B349-4CFC-A49E-A1AEA43FDD79) Beginning test session XCUITests-A8CF1B79-B349-4CFC-A49E-A1AEA43FDD79 at 2020-08-31 23:19:44.646 with Xcode 11C505 on target <DVTiPhoneSimulator: 0x7fd0d9960000> {
		SimDevice: iPhone 11 Pro (70010BF3-FDED-4165-9813-C67BA2CF6A9B, iOS 13.3, Booted)
} (13.3 (17C45))
2020-08-31 23:20:17.908 xcodebuild[2145:36022] [MT] IDETestOperationsObserverDebug: (A8CF1B79-B349-4CFC-A49E-A1AEA43FDD79) Finished requesting crash reports. Continuing with testing.
2020-08-31 23:20:23.155630+0000 XCUITests-Runner[2184:52430] Running tests...
Test Suite 'All tests' started at 2020-08-31 23:20:25.423
Test Suite 'XCUITests.xctest' started at 2020-08-31 23:20:25.424
Test Suite 'BuckSampleUITests' started at 2020-08-31 23:20:25.424
Test Case '-[XCUITests.BuckSampleUITests testExampleAppLabel]' started.
    t =     0.00s Start Test at 2020-08-31 23:20:25.425
    t =     0.10s Set Up
    t =     0.10s     Open com.airbnb.ExampleApp
    t =     0.17s         Launch com.airbnb.ExampleApp
    t =     0.29s             Setting up automation session
    t =     2.63s             Wait for com.airbnb.ExampleApp to idle
    t =     4.02s Find the Target Application 'com.airbnb.ExampleApp'
    t =     4.04s Tear Down
Test Case '-[XCUITests.BuckSampleUITests testExampleAppLabel]' passed (4.250 seconds).
Test Case '-[XCUITests.BuckSampleUITests testMainScreenLabelExists]' started.
    t =     0.00s Start Test at 2020-08-31 23:20:29.676
    t =     0.05s Set Up
    t =     0.05s     Open com.airbnb.ExampleApp
    t =     0.10s         Launch com.airbnb.ExampleApp
    t =     0.10s             Terminate com.airbnb.ExampleApp:2186
    t =     1.33s             Setting up automation session
    t =     3.65s             Wait for com.airbnb.ExampleApp to idle
    t =     4.74s Checking existence of `"Hello, world" StaticText`
    t =     4.76s Tear Down
Test Case '-[XCUITests.BuckSampleUITests testMainScreenLabelExists]' passed (4.967 seconds).
Test Suite 'BuckSampleUITests' passed at 2020-08-31 23:20:34.643.
	 Executed 2 tests, with 0 failures (0 unexpected) in 9.217 (9.218) seconds
Test Suite 'XCUITests.xctest' passed at 2020-08-31 23:20:34.643.
	 Executed 2 tests, with 0 failures (0 unexpected) in 9.217 (9.219) seconds
Test Suite 'All tests' passed at 2020-08-31 23:20:34.644.
	 Executed 2 tests, with 0 failures (0 unexpected) in 9.217 (9.221) seconds
```

@shepting @qyang-nj

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,11 +52,18 @@ test:
...
 ui_test:
-	# Diable UI Test for now, because it's broken on Xcode 10.2
-	# $(BUCK) test //App:XCUITests --config apple.xctool_path=$(fbxctest)
+	$(BUCK) build //App:XCUITests
+	mkdir -p ${UI_TESTS_TMP}
+	ln -sf $(buck_out)/gen/App/XCUITests#apple-test-bundle,dwarf,no-include-frameworks,no-linkermap/XCUITests.xctest $(UI_TESTS_TMP)
+	cp $(UI_TESTS_TOOLS)/ExampleApp.xctestrun $(UI_TESTS_TMP)
+	rm -rf $(UI_TESTS_TMP)/XCUITests-Runner.app
+	unzip $(UI_TESTS_TOOLS)/XCUITests-Runner.app.zip -d $(UI_TESTS_TMP)
+	xcrun simctl boot "iPhone 11 Pro"
```
Comment (qyang-nj) [inline Makefile:64]: This will fail if the sim is already booted. It's better to check its status or shut it down before booting.

File: Makefile [hunk 2]
```diff
@@ -52,11 +52,18 @@ test:
...
+UI_TESTS_TMP = $(shell $(BUCK) root)/build/xcuitest
+UI_TESTS_TOOLS = $(shell $(BUCK) root)/tools/xcuitest
 ui_test:
-	# Diable UI Test for now, because it's broken on Xcode 10.2
-	# $(BUCK) test //App:XCUITests --config apple.xctool_path=$(fbxctest)
+	$(BUCK) build //App:XCUITests
+	mkdir -p ${UI_TESTS_TMP}
+	ln -sf $(buck_out)/gen/App/XCUITests#apple-test-bundle,dwarf,no-include-frameworks,no-linkermap/XCUITests.xctest $(UI_TESTS_TMP)
+	cp $(UI_TESTS_TOOLS)/ExampleApp.xctestrun $(UI_TESTS_TMP)
+	rm -rf $(UI_TESTS_TMP)/XCUITests-Runner.app
```
Comment (qyang-nj) [inline Makefile:62]: Where does `XCUITests-Runner.app` come from? Was it left from previous run? If so, it's probably better to `rm -rf $(UI_TESTS_TMP)` at the beginning of this task.

File: Makefile [hunk 3]
```diff
@@ -52,11 +52,18 @@ test:
...
+	unzip $(UI_TESTS_TOOLS)/XCUITests-Runner.app.zip -d $(UI_TESTS_TMP)
+	xcrun simctl boot "iPhone 11 Pro"
+	xcrun simctl install "iPhone 11 Pro" $(UI_TESTS_TMP)/XCUITests.xctest/PlugIns/ExampleApp.app
+	xcodebuild test-without-building -xctestrun $(UI_TESTS_TMP)/ExampleApp.xctestrun -destination 'platform=iOS Simulator,name=iPhone 11 Pro,OS=latest'
```
Comment (qyang-nj) [inline Makefile:66]: I hope `iPhone 11 Pro` can be a variable, since it's used three times here.

File: Makefile (hunk 1 above)
Comment (xianwen) [reply Makefile:64]: `xcrun simctl shutdown` would also fail if it had already been shutdown, I'll use a simple `|| true` here to ignore the error if it's already booted.

**Comment by xianwen** [comment]
@qyang-nj , PTAL :)

**Review by qyang-nj — APPROVED** [review]
🙌

**Pull request merged**